.changeset/cold-grapes-collect.md

```diff
@@ -0,0 +1,32 @@
+---
+"@aws-amplify/ui-react-native": patch
+"@aws-amplify/ui-react": patch
+"@aws-amplify/ui": patch
+---
+
+fix(react-native): border widths, spacing, font sizes, opacities in the theme don't throw runtime errors.
+
+These are all valid in a theme now:
+
+```typescript
+const theme: Theme = {
+  tokens: {
+    borderWidths: {
+      small: '4',
+      medium: '1rem',
+      large: 6,
```

Comment:
Is this something specific in RN where this works without a unit? Or do we just not want it to break completely if people try to do it

Reply:
Specific to RN because when we use the theme we use it in a RN StyleSheet which is typed rather than outputting to a CSS string which is untyped. Most numeric values (space, borderWidth, fontSize, etc.) are expected to be a number rather than a string. In CSS its all just strings and numbers can be cast to strings but not the other way around.